src/main/scala/com/databricks/labs/overwatch/ApiCallV2.scala

```diff
@@ -0,0 +1,549 @@
+  private var _successTempPath: String = _ //Unique String which is used as folder name in a temp location to save the responses.
+  private var _unsafeSSLErrorCount = 0; //Keep track of SSL error occurrence.
+  private var _apiMeta: ApiMeta = null //Metadata for the API call.
+  private var _debugFlag = false //Debug flag to print the information when required. d
+  private var _allowUnsafeSSL = false //Flag to make the unsafe ssl.
+  private var _jsonKey: String = "" //Key name for pagination.
+  private var _jsonValue: String = "" //Key value for pagination.
+  private val readTimeoutMS = 60000 //Read timeout.
+  private val connTimeoutMS = 10000 //Connection timeout.
+  private var _printFlag: Boolean = true
+
+  def endPoint: String = _endPoint
+
+  def jsonQuery: String = _jsonQuery
+
+  def serverBusyCount: Int = _serverBusyCount
+
+  def printFlag: Boolean = _printFlag
+
+  def token: String = _token
+
+  def successTempPath: String = _successTempPath
+
+  def unsafeSSLErrorCount: Int = _unsafeSSLErrorCount
+
+  def apiMeta: ApiMeta = _apiMeta
+
+  def debugFlag: Boolean = _debugFlag
+
+  def allowUnsafeSSL: Boolean = _allowUnsafeSSL
+
+  def jsonKey: String = _jsonKey
+
+  def jsonValue: String = _jsonValue
+
+  def apiResponseArray: util.ArrayList[String] = _apiResponseArray
+
+  private[overwatch] def setApiResponseArray(value: util.ArrayList[String]): this.type = {
+    _apiResponseArray = value
+    this
+  }
+
+  private[overwatch] def setJsonValue(value: String): this.type = {
+    _jsonValue = value
+    this
+  }
+
+  private[overwatch] def setJsonKey(value: String): this.type = {
+    _jsonKey = value
+    this
+  }
+
+  private[overwatch] def setAllowUnsafeSSL(value: Boolean): this.type = {
+    _allowUnsafeSSL = value
+    this
+  }
+
+  private[overwatch] def setDebugFlag(value: Boolean): this.type = {
+    _debugFlag = value
+    this
+  }
+
+  private[overwatch] def setApiMeta(value: ApiMeta): this.type = {
+    _apiMeta = value
+    this
+  }
+
+  private[overwatch] def setUnsafeSSLErrorCount(value: Int): this.type = {
+    _unsafeSSLErrorCount = value
+    this
+  }
+
+  private[overwatch] def setSuccessTempPath(value: String): this.type = {
+    _successTempPath = value
+    this
+  }
+
+  private[overwatch] def setToken(value: String): this.type = {
+    _token = value
+    this
+  }
+
+  private[overwatch] def setEndPoint(value: String): this.type = {
+    _endPoint = value
+    this
+  }
+
+  private[overwatch] def setJsonQuery(value: String): this.type = {
+    _jsonQuery = value
+    this
+  }
+
+  private[overwatch] def setPrintFlag(value: Boolean): this.type = {
+    _printFlag = value
+    this
+  }
+
+  private[overwatch] def setServerBusyCount(value: Int): this.type = {
+    _serverBusyCount = value
+    this
+  }
+
+  /**
+   * Setting up the api name and api metadata for that api.
+   *
+   * @param value
+   * @return
+   */
+  private def setApiName(value: String): this.type = {
```

Comment:
can we rename this? This is the primary entry / build point, the constructor for the class essentially...how about `buildAPI`?